### DISCUSSION TITLE
Variable scale

### DISCUSSION THREAD

**Discussion by TheSimplyRed** [coin-or/Ipopt#490] [Q&A] [answered]
Hi,
If I have one variable between 500 and 5000, and all others are between 0 and 100, Do I have to scale this variable or it is ok if I use it as it is?

Thanks

**Comment by svigerske** [comment, marked as answer]
Could be ok.

If you have HSL MC19, then setting nlp_scaling_method to equilibration-based would tell Ipopt to figure out scaling for both variables and constraints: https://coin-or.github.io/Ipopt/OPTIONS.html#OPT_nlp_scaling_method

**Reply by TheSimplyRed** [reply]
I did a mistake variables are between 0 and 1 not 0 and 100. Is it still ok if one variable is between 500 and 5000, or I need to scale it? I'm using Mumps.
Is ipopt able to find variables arround  1e-70? 
Thanks

**Reply by svigerske** [reply]
Just try out whether it works. I don't think that one could give a definite answer in general.

Variables can take any floating point value, so 1e-70 is possible.
However, the precision of floating point arithmetics sets some limits on the accuracy of solutions that one can expect.
